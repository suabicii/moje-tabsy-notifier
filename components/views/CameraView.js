import React from "react";
import {View} from "react-native";
import {BarCodeScanner} from "expo-barcode-scanner";
import BarcodeMask from "react-native-barcode-mask";

function CameraView({hasPermission, scanned, handleBarcodeScanned}) {
    return (
        <>
            {
                hasPermission === true
                    ?
                    <View
                        testID="camera-view"
                        style={{
                            aspectRatio: 1,
                            borderRadius: 10,
                            marginBottom: 20,
                            marginLeft: 25
                        }}
                    >
                        <BarCodeScanner
                            testID="barcode-scanner"
                            style={{flex: 1}}
                            onBarCodeScanned={scanned ? undefined : handleBarcodeScanned}
                        />
                        <BarcodeMask edgeColor="#62B1F6" showAnimatedLine animatedLineColor="#FF0000"/>
                    </View>
                    :
                    <></>
            }
        </>
    );
}

export default CameraView;