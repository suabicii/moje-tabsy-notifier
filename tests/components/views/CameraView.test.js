import React from "react";
import renderer from "react-test-renderer";
import CameraView from "../../../components/views/CameraView";

it('should correctly render CameraView with granted permission', () => {
    const tree = renderer.create(
        <CameraView
            hasPermission={true}
            scanned={true}
            handleBarcodeScanned={jest.fn()}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});