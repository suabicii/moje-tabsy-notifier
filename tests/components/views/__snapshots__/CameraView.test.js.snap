// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render CameraView with granted permission 1`] = `
<View
  style={
    {
      "aspectRatio": 1,
      "borderRadius": 10,
      "marginBottom": 20,
      "marginLeft": 25,
    }
  }
  testID="camera-view"
>
  <ViewManagerAdapter_ExpoBarCodeScanner
    barCodeTypes={[]}
    onBarCodeScanned={[Function]}
    style={
      {
        "flex": 1,
      }
    }
    testID="barcode-scanner"
  />
  <View
    style={
      [
        {
          "alignItems": "center",
          "bottom": 0,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        [
          {
            "alignItems": "center",
            "justifyContent": "center",
          },
          {
            "height": 230,
            "width": 280,
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "borderColor": "#62B1F6",
              "height": 20,
              "width": 20,
            },
            {
              "left": 0,
              "position": "absolute",
              "top": 0,
            },
            {
              "borderLeftWidth": 4,
              "borderTopLeftRadius": undefined,
              "borderTopWidth": 4,
              "left": 0,
              "top": 0,
            },
          ]
        }
      />
      <View
        style={
          [
            {
              "borderColor": "#62B1F6",
              "height": 20,
              "width": 20,
            },
            {
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            {
              "borderRightWidth": 4,
              "borderTopRightRadius": undefined,
              "borderTopWidth": 4,
              "right": 0,
              "top": 0,
            },
          ]
        }
      />
      <View
        style={
          [
            {
              "borderColor": "#62B1F6",
              "height": 20,
              "width": 20,
            },
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
            },
            {
              "borderBottomLeftRadius": undefined,
              "borderBottomWidth": 4,
              "borderLeftWidth": 4,
              "bottom": 0,
              "left": 0,
            },
          ]
        }
      />
      <View
        style={
          [
            {
              "borderColor": "#62B1F6",
              "height": 20,
              "width": 20,
            },
            {
              "bottom": 0,
              "position": "absolute",
              "right": 0,
            },
            {
              "borderBottomRightRadius": undefined,
              "borderBottomWidth": 4,
              "borderRightWidth": 4,
              "bottom": 0,
              "right": 0,
            },
          ]
        }
      />
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#FF0000",
            "elevation": 4,
            "height": 2,
            "margin": 4,
            "maxHeight": 230,
            "maxWidth": 280,
            "position": "absolute",
            "width": "85%",
            "zIndex": 0,
          }
        }
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "height": "100%",
          "justifyContent": "space-around",
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        }
      }
    >
      <View
        style={
          [
            {
              "width": "100%",
            },
            {
              "backgroundColor": "rgb(0, 0, 0)",
              "flex": 1,
              "opacity": 0.6,
            },
          ]
        }
      />
      <View
        style={
          [
            {
              "height": 230,
            },
            {
              "display": "flex",
              "flexDirection": "row",
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "backgroundColor": "rgb(0, 0, 0)",
                "flex": 1,
                "opacity": 0.6,
              },
            ]
          }
        />
        <View
          style={
            [
              {
                "backgroundColor": "transparent",
              },
              {
                "height": 230,
                "width": 280,
              },
            ]
          }
        />
        <View
          style={
            [
              {
                "backgroundColor": "rgb(0, 0, 0)",
                "flex": 1,
                "opacity": 0.6,
              },
            ]
          }
        />
      </View>
      <View
        style={
          [
            {
              "width": "100%",
            },
            {
              "backgroundColor": "rgb(0, 0, 0)",
              "flex": 1,
              "opacity": 0.6,
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
